SYCL/ESIMD/api/functional/functions/functions_atomic_update_inc_dec.cpp

```diff
@@ -0,0 +1,239 @@
+
+using namespace esimd_test::api::functional;
+using namespace sycl::ext::intel::esimd;
+
+// Descriptor class for the case of calling atomic_update overload for zero
+// operands.
+struct atomic_update_0_operands {
+  static std::string get_description() {
+    return "atomic update with zero operands";
+  }
+
+  template <typename DataT, int NumElemsToChange, typename ChangeElemFilterT,
+            atomic_op ChosenOperator>
+  static int call_esimd_function(DataT *input_data, size_t *const offsets) {
+    simd<DataT, NumElemsToChange> offset;
+    mask_type_t<NumElemsToChange> mask;
+
+    ChangeElemFilterT filter{};
+    for (size_t i = 0; i < NumElemsToChange; ++i) {
+      mask[i] = filter(i);
+      offset[i] = offsets[i] * sizeof(DataT);
+    }
+
+    auto values_before_update =
+        atomic_update<ChosenOperator, DataT, NumElemsToChange>(input_data,
+                                                               offset, mask);
+
+    unsigned int values_sum = 0;
+    for (size_t i = 0; i < NumElemsToChange; ++i) {
+      values_sum += values_before_update[i];
+    }
+
+    return values_sum;
+  }
+};
+
+// Provides expected value for updated memory locations.
+template <esimd::atomic_op Operator, typename T>
+T get_expected_value(T base_value, int number_of_iterations) {
+  if constexpr (Operator == esimd::atomic_op::dec) {
+    return base_value - number_of_iterations;
+  } else if constexpr (Operator == esimd::atomic_op::inc) {
+    return base_value + number_of_iterations;
+  } else {
+    static_assert(Operator != Operator, "Unexpected  operator type.");
+  }
+}
+
+// The main test routine.
+// Using functor class to be able to iterate over the pre-defined data types.
+template <typename TestCaseT, typename DataT, typename NumElemsT,
+          typename NDRangeDimT, typename NumElemsToChangeT,
+          typename ChangeElemFilterT, typename ChosenOperatorT,
+          typename AlgorithmToChangeT>
+class run_test {
+  static constexpr int NumElems = NumElemsT::value;
+  static constexpr int NDRangeDim = NDRangeDimT::value;
+  static constexpr int NumElemsToChange = NumElemsToChangeT::value;
+  static constexpr atomic_op ChosenOperator = ChosenOperatorT::value;
+  static constexpr functions::offset_generation AlgorithmToChange =
+      AlgorithmToChangeT::value;
+  using TestDescriptionT = functions::TestDescription<NumElems, TestCaseT>;
+
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type,
+                  const std::string &operator_type) {
+    static_assert(NDRangeDim <= 3 && NDRangeDim > 0);
+    static_assert(NumElemsToChange <= NumElems);
+    bool passed = true;
+    constexpr DataT base_value = 100;
+    constexpr DataT result_fill_value = 5;
+    shared_allocator<DataT> allocator(queue);
+    shared_vector<DataT> shared_init_values(NumElems, allocator);
+    functions::fill_init_values<NumElems, ChosenOperator, DataT>(
+        base_value, shared_init_values);
```

Comment:
```suggestion
    std::fill(shared_init_values.begin(), shared_init_values.end(), base_value)
```
By doing so you make explicit statement what we can expect after increment/decrement operators.
So it's easier to read verification code below.

Reply:
I've used `std::fill` here [here](https://github.com/intel/llvm-test-suite/pull/939/commits/43ce07f5132c624c1f20ab064dbaf302ce9788e9#diff-afaea17121d6706af52229dcfb462795dfe86fde5c98f09b836505a741830b6bL102-R105)